### PR TITLE
Add ability to solve() to a given pattern

### PR DESCRIPTION
**Pull request by ansiwen**
Second attempt, this time with C implementation as well. Solves #6 and #13. Replaces #8.

### CODE REVIEW

**Comment by ansiwen** [comment]
What are the policies for master vs. develop branch? Do you want new things in general be committed into develop branch now? Then I can rebase this.

**Comment by muodov** [comment]
I prefer doing it "git flow"-way. So `develop` is the main working branch, and all `feature-X` branches get merged there. To create a new release, I start a `release-N` branch based on `develop`, update changelog/readme/etc and then merge it into `develop` and `master` with a tag. So `master` always contains the latest "stable" release, and `develop` contains a "nightly" state.

File: kociemba/__init__.py [hunk 1]
```diff
@@ -9,8 +9,9 @@ class SlowContextWarning(Warning):
     from .ckociembawrapper import ffi, lib
     cache_dir = os.path.join(os.path.dirname(__file__), 'cprunetables')
 
-    def _solve(s):
-        res = lib.solve(s.encode('utf-8'), cache_dir.encode('utf-8'))
+    def _solve(cube, pattern):
+        pattern_utf8 = pattern.encode('utf-8)') if pattern is not None else ffi.NULL
```
Comment (muodov) [inline kociemba/__init__.py]: there is a small typo here in encoding name

**Comment by muodov** [comment]
Looks good to me except for that typo. Just create a PR against `develop` branch and I'll merge it. Too bad [this](https://github.com/isaacs/github/issues/18) feature is still not available in GitHub.

**Pull request closed without merging**